sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -225,7 +225,7 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev)
 
 	/* clear WAKESTS */
 	snd_sof_dsp_write(sdev, HDA_DSP_HDA_BAR, SOF_HDA_WAKESTS,
-			  SOF_HDA_WAKESTS_INT_MASK);
+			  bus->codec_mask);
```

Comment:
Do we want to take the opportunity to clear only SDW codecs WAKESTS in intel_shim_wake()?
I.e.
Change
```
		/* Clear wake status (W1C) */
		wake_sts = snd_hdac_chip_readw(sdw->link_res->hbus, STATESTS);
		wake_sts |= lsdiid;
		snd_hdac_chip_writew(sdw->link_res->hbus, STATESTS, wake_sts);
```
to
```
		snd_hdac_chip_writew(sdw->link_res->hbus, STATESTS, lsdiid);
```

Reply:
yep, that sounds good.

Those W1C bits are nasty, when trying to do the right thing with a read-modify-write you end-up adding side effects without realizing it...

Reply:
done